File: backend/src/main/scala/domains/blog/api/VoteBlogCommentInput.scala
package domains.blog.api

import domains.blog.objects.{BlogCommentId, BlogId}
import domains.blog.objects.request.VoteBlogCommentRequest

private[api] final case class VoteBlogCommentInput(
  blogId: BlogId,
  commentId: BlogCommentId,
  request: VoteBlogCommentRequest
)

File: frontend/src/objects/blog/request/VoteBlogCommentInput.ts
import type { BlogCommentId } from '@/objects/blog/BlogCommentId'
import type { BlogId } from '@/objects/blog/BlogId'
import type { VoteBlogCommentRequest } from '@/objects/blog/request/VoteBlogCommentRequest'

/** 博客评论投票输入；路径参数定位评论，请求体携带投票方向。 */
export type VoteBlogCommentInput = {
  blogId: BlogId
  commentId: BlogCommentId
  request: VoteBlogCommentRequest
}
